tests/generators/cpp/src/test.cpp

```diff
@@ -14,3 +17,337 @@
+  void setProperty(Args&&... args) {
+    trait.setBoolProperty(std::forward<Args>(args)...);
+  }
+};
+
+template <>
+struct PropertyFixture<openassetio::Int> {
+  using AllPropertiesTrait = openassetio_traitgen_test_all::traits::aNamespace::AllPropertiesTrait;
+
+  AllPropertiesTrait trait;
+
+  static inline const openassetio::trait::property::Key kPropertyKey = "intProperty";
+  static constexpr std::string_view kPropertyType = "openassetio::Int";
+  static constexpr openassetio::Int kDefaultValue = 456;
+  static constexpr openassetio::Int kExpectedValue = 123;
+  static constexpr openassetio::Bool kUnexpectedValue = true;
+
+  template <typename... Args>
+  [[nodiscard]] auto getProperty(Args&&... args) const {
+    return trait.getIntProperty(std::forward<Args>(args)...);
+  }
+
+  template <typename... Args>
+  void setProperty(Args&&... args) {
+    trait.setIntProperty(std::forward<Args>(args)...);
+  }
+};
+
+template <>
+struct PropertyFixture<openassetio::Float> {
+  using AllPropertiesTrait = openassetio_traitgen_test_all::traits::aNamespace::AllPropertiesTrait;
+
+  AllPropertiesTrait trait;
+
+  static inline const openassetio::trait::property::Key kPropertyKey = "floatProperty";
+  static constexpr std::string_view kPropertyType = "openassetio::Float";
+  static constexpr openassetio::Float kDefaultValue = 12.3;
+  static constexpr openassetio::Float kExpectedValue = 45.6;
+  static constexpr openassetio::Int kUnexpectedValue = 123;
+
+  template <typename... Args>
+  [[nodiscard]] auto getProperty(Args&&... args) const {
+    return trait.getFloatProperty(std::forward<Args>(args)...);
+  }
+
+  template <typename... Args>
+  void setProperty(Args&&... args) {
+    trait.setFloatProperty(std::forward<Args>(args)...);
+  }
+};
+
+template <>
+struct PropertyFixture<openassetio::Str> {
+  using AllPropertiesTrait = openassetio_traitgen_test_all::traits::aNamespace::AllPropertiesTrait;
+
+  AllPropertiesTrait trait;
+
+  static inline const openassetio::trait::property::Key kPropertyKey = "stringProperty";
+  static constexpr std::string_view kPropertyType = "openassetio::Str";
+  static inline const openassetio::Str kDefaultValue = "⛅ outside today";
+  static inline const openassetio::Str kExpectedValue = "🐁";
+  static constexpr openassetio::Int kUnexpectedValue = 123;
+
+  template <typename... Args>
+  [[nodiscard]] auto getProperty(Args&&... args) const {
+    return trait.getStringProperty(std::forward<Args>(args)...);
+  }
+
+  template <typename... Args>
+  void setProperty(Args&&... args) {
+    trait.setStringProperty(std::forward<Args>(args)...);
+  }
+};
+}  // namespace
+
+TEMPLATE_TEST_CASE("Property getters", "", openassetio::Bool, openassetio::Int, openassetio::Float,
+                   openassetio::Str) {
+  using openassetio_traitgen_test_all::traits::aNamespace::AllPropertiesTrait;
+
+  using PropertyType = TestType;  // Catch2-injected template param.
+  using Fixture = PropertyFixture<PropertyType>;
```

Comment:
[Consider] I wonder if the complexity of the template specializations are necessary here. I may be missing something, but I speculate you could define the structs as regular `FloatPropertyFixture`, `StringPropertyFixure`, etc, and simply have a little switch-statement dispatcher.

The only benefit of static dispatch that might apply here, given that you have no non-specialized generic case, is avoiding the minor performance overhead of runtime branching, which I don't think matters given this is test code.

Reply:
Good point, if I understood you correctly.  My previous experience using `TEMPLATE_TEST_CASE` made me default to viewing the `TestType` as a template parameter, rather than the final type, which it certainly could be.  We could totally have separate fixture classes, rather than template specialisations.

Though not sure what you mean by a `switch` statement?  I think the code would be near-identical, other than the line you highlighted?

Unless I've misunderstood.  Are you thinking we could factor out the assertions into e.g. a generic lambda, then conditionally call the lambda with different fixtures to test the different methods?

Reply:
I thought that if you remove the template specializations, you would need to add a `BuildFixture(FixtureType)` sort of function, to create the fixtures of the appropriate type in the functions, which I guessed would just be a switch/chained-ifs statement over the types (OpenAssetIO::bool, etc).

I'm hanging onto my understanding of what's happening here pretty loosely, so don't take anything i'm saying as gospel :P

Reply:
I guess maybe test test framework dosen't provide a runtime variable like that, which is why these are all specializations in the first place?

Reply:
Yeah there is no runtime variable, just a compile-time type that you can use to branch/parametrise your tests.

Reply:
Ah well, I guess disregard this entire thread then.